### PR TITLE
DEV2-2532 - refactor enterprise v1

### PR DESCRIPTION
**Pull request by dimacodota**
this is the first iteration of the refactoring, this pr refactors the enterprise dependencies and adds the updating capabilities to the enterprise build:
on top of this pr two additional PRs will be opened:
  1 - move all the shared logic to a "shared" folder
  2 - separate the extension/shared/enterprise to different packages

### CODE REVIEW

File: .github/workflows/package-enterprise.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16.19'
+      - name: Tag Version
+        id: tag-version
+        run: echo ::set-output name=RELEASE_VERSION::$(echo ${GITHUB_REF:10})
```
Comment (assapir) [inline .github/workflows/package-enterprise.yml:23]: Use the new syntax for outputs

File: .github/workflows/package-enterprise.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        run: |
+          echo "" > ./README.md
+          echo "buildId: ${{ github.sha }}" >> ./README.md
+          echo "" >> ./README.md
+
+      - name: Install
+        run: yarn install --frozen-lockfile --ignore-optional
+
+      - name: Modify package
+        run: npm pkg set \"name=tabnine-vscode-enterprise\" && npm pkg delete __metadata
```
Comment (assapir) [inline .github/workflows/package-enterprise.yml:35]: why deleting `__metadata`?

File: src/enterprise/update/client.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import { workspace } from "vscode";
+import * as Url from "url";
+import axios, { AxiosInstance } from "axios";
+import * as http from "http";
+import * as https from "https";
+
+function proxyUrl(): string | undefined {
+  return (
+    workspace.getConfiguration().get<string>("http.proxy") ||
+    process.env.HTTPS_PROXY ||
```
Comment (assapir) [inline src/enterprise/update/client.ts:10]: where is `no_proxy` got lost?

File: src/globals/tabnineExtensionProperties.ts [hunk 4]
```diff
@@ -175,6 +181,14 @@ function getContext(): TabNineExtensionProperties {
       }
       return undefined;
     },
+    get shouldRunBootstrap(): boolean {
+      return !cloudHost;
```
Comment (assapir) [inline src/globals/tabnineExtensionProperties.ts:185]: I think it's better to do that based if enterprise, not if cloudHost is configured

File: package.json [hunk 5]
```diff
@@ -231,10 +231,6 @@
         "command": "TabNine::assistantToggle",
         "title": "Tabnine: Toggle Assistant"
       },
-      {
```
Comment (assapir) [inline package.json]: YES THANK YOU

**Review by amircodota — APPROVED** [review]
LGTM

**Pull request merged**